### PR TITLE
Adding tooltips and examine hints to window frames (plus some other bs)

### LINKED ISSUE #85769: Window Frames do not have a tooltip for how to deconstruct (and they drop nothing) (Wallening)
Reporting client version: 515.1641

## Issue Summary
Windowframe

You must use a welder to deconstruct a window frame, but examine text does not tell you this so it's a bit confusing. You will also receive nothing upon it being welded down.

## Round ID:
[234562](https://scrubby.melonmesa.com/round/234562)

## Testmerges:
- [[tm]Revert "Refactor Stat Panel styles and more TGUI-like appearance"](https://www.github.com/tgstation/tgstation/pull/85392)
- [Wallening [IDB IGNORE] [MDB IGNORE]](https://www.github.com/tgstation/tgstation/pull/85491)

## Reproduction:

Find a window sill, weld it with right click, observe as nothing comes out of it and that you needed me to tell you this to find out that's how you can deconstruct it.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This will fix #85769, plus we retain the type of rods used for building the grille.

## Why It's Good For The Game
This will fix #85769.

## Changelog

:cl:
add: Added tooltips and examine hints to window frames.
fix: Fixed deconstructing window frames not dropping sheets, and rods if a grille was installed.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/window_frames.dm [hunk 1]
```diff
@@ -54,17 +54,47 @@
 /obj/structure/window_frame/Initialize(mapload)
 	. = ..()
 
+	if(grille)
+		grille = new grille(src, sheet_amount)
```
Comment (SmArtKar) [inline code/game/objects/structures/window_frames.dm:58]: you'd probably want to istype this just for clarity

**Review by SmArtKar — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/window_frames.dm (hunk 1 above)
Comment (Ghommie) [reply code/game/objects/structures/window_frames.dm:58]: What?

File: code/game/objects/structures/window_frames.dm [hunk 2]
```diff
@@ -170,6 +207,13 @@
 
 	return ITEM_INTERACT_SUCCESS
 
+/obj/structure/window_frame/handle_deconstruct(disassembled)
+	if(!disassembled)
+		grille?.use(round(sheet_amount * 0.5))
```
Comment (ArcaneMusic) [inline code/game/objects/structures/window_frames.dm:212]: This runtimes in testing.
`Runtime in code/game/objects/structures/window_frames.dm, line 212: Cannot execute 0.use().`

Same goes for 213 when deconstructing a frame:
`code/game/objects/structures/window_frames.dm, line 213: Cannot execute 0.forceMove().`

File: code/game/objects/structures/window_frames.dm [hunk 3]
```diff
@@ -54,17 +54,47 @@
...
+		TOOL_WELDING = list(
+			SCREENTIP_CONTEXT_LMB = "Repair",
+			SCREENTIP_CONTEXT_RMB = "Deconstruct",
+		),
+		TOOL_WIRECUTTER = list(
+			SCREENTIP_CONTEXT_LMB = "Remove grille",
+		),
+
+	)
+	AddElement(/datum/element/contextual_screentip_tools, tool_tips)
```
Comment (ArcaneMusic) [inline code/game/objects/structures/window_frames.dm:74]: Screentips on both the Tool-side and for the iron rod/grill sides of things are not showing up in testing. Glass however is showing up correctly.

File: code/game/objects/structures/window_frames.dm [hunk 4]
```diff
@@ -225,28 +269,52 @@
...
-			has_grille = TRUE
-			to_chat(user, "<span class='notice'>You add [stack_name] to [src]")
-			update_appearance()
-
-	else if((attacking_item.obj_flags & CONDUCTS_ELECTRICITY) && try_shock(user, 70))
+
+	if(!isstack(attacking_item))
+		if((attacking_item.obj_flags & CONDUCTS_ELECTRICITY) && try_shock(user, 70))
+			return
+	return ..()
```
Comment (ArcaneMusic) [inline code/game/objects/structures/window_frames.dm:276]:
```suggestion
		return ..()
```
Should fix much of the issues with rods not building on the frame

**Review by ArcaneMusic — CHANGES_REQUESTED** [review]

**Comment by Ghommie** [comment]
Wallening is going to get reverted so...

**Pull request closed without merging**